Tests/mocks/devicesettings.h

```diff
@@ -1215,6 +1228,399 @@ class FrontPanelConfig {
         return impl->getTextDisplay();
     }
 };
-
 }
 
+
+typedef uint32_t Op_t;
+class Exception : public std::exception
+{
+	public:
+		virtual const char* what() const throw()
+		{
+			return "Base Exception..";
+		}
+
+};
+class CECNoAckException : public Exception
+{
+	public:
+		virtual const char* what() const throw()
+		{
+			return "Acl not received..";
+		}
+};
+class IOException : public Exception
+{
+	public:
+		virtual const char* what() const throw()
+		{
+			return "IO Exception..";
+		}
+};
+
+class CECFrameImpl{
+	public:
+		virtual void getBuffer(const uint8_t **buf, size_t *len) const = 0;
+
+};
+class CECFrame{
+	public:
+	
+	CECFrameImpl* impl;
+	void getBuffer(const uint8_t **buf, size_t *len) const{
+		return impl->getBuffer(buf, len);
+	}
+	CECFrame(const uint8_t *buf = NULL, uint16_t len = 0){}
+
+};
+class CECBytesImpl {
+	public:
+	virtual const std::string toString() const = 0;
+};
+class CECBytes{
+    public:
+    CECBytesImpl* impl;
+    static CECBytes& getInstance()
```

Comment:
if code creates CECBytes object and calls toString on it that will trigger object's impl, not CECBytes::getInstance().impl. No service code uses CECBytes::getInstance(). No static methods on CECBytes that need a static mock. So getInstance() is not needed. Also, I can't see any usages of CECBytes in project. Is it needed?

Reply:
@npoltorapavlo So, in the base code, several classes inherit from CECBytes, which is the class that has toString(). There are several classes that are needed, but not used (because they are called through the notify function, which is private, and not called directly within the code. Notify() does get called in actual through some weird use of inheritance and overriding). Using the inheritance of CECBytes, I can have the classes that don't get called and not throw errors without needed to create additional mock files that won't be used. I can revert to not using CECBytes, if its preferred.

Reply:
ok, but how does it work if you set mock on a static CECBytes (`CECBytes::getInstance().impl = &cecBytesImplMock;`) and in CECBytes::toString(void) you call impl->toString(); and not getInstance().impl->toString();?

Reply:
Oh, I see what you mean. You're right, it should be getInstance().impl->toString.